### PR TITLE
generate Api for my backend

### PR DESCRIPTION
**Pull request by osbentech**
In this  branch I have implemented the following:
    - Create a new Rails API app called 'hello-rails-back-end'.
    - Initialize your project with Git.
    -  Make sure that your project has a Postgres database set up.
    - Create a table for storing your messages and create 5 different greetings.
    - Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and rails route).

    - Create a new React app called 'hello-react-front-end'
    - Initialize your project with Git.
    - Create the App component with react-router.
    - Create the Greeting component that will display a greeting. Set it up as a route in your App component.
    - Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
    - Display the random greeting in your Greeting component.

### CODE REVIEW

**Comment by MahmoudBakr23** [comment]
### Hi @osbentech 😃, 

As much as I appreciate your hard work on this project, I'm afraid that I can't perform a review on this pull request.

**Invalid Submission**
- Please attach the link to the front-end repo in the README file and this PR summary, and do the other way around in the front-end's repo README file and PR as well. 🙏 

_Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready._ 🙏

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

<hr>

_Invalid Code Review Request does not count into the code reviews limit._

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @osbentech 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- For the frontend project, you ought to create a pull request from a `feature` branch to the `dev` branch. Currently, all your changes are being made on the `dev` branch. Kindly correct this 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/seeds.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This file should contain all the record creation needed to seed the database with its default values.
+# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
+#
+# Examples:
+#
+#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
+#   Character.create(name: "Luke", movie: movies.first)
```
Comment (franclobo) [inline db/seeds.rb:7]: - **[OPTIONAL]** I recommend loading the data base using db:seed to have greetings to show 👇🏽 

```ruby
Message.create!(message: "Hello")
Message.create!(message: "Hi!")
Message.create!(message: "Hola")
Message.create!(message: "Bonjour")
Message.create!(message: "Ciao") 
```

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Rails.application.routes.draw do
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  # Defines the root path route ("/")
+  namespace :api do
+    namespace :v1 do
+      resources :roots
+      get '/message', to: 'messages#index'
+    end
+  end
+end
```
Comment (franclobo) [inline config/routes.rb:10]: - [ ] Please add the root to the index page to load the greetings, not to the default page (Rails logo).

`root to: 'api/v1/roots#index'`

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @osbentech, 

terminator

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have the files changed in the front-end PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by prolajumokeoni** [comment]
Hi @osbentech 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have linters setup, please add the linters config in your frontend repository 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by arslanbisharat** [comment]
Hi @osbentech , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a PR without proper description. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by afanimartin — APPROVED** [review]
Hi @osbentech,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Route for random messages added to the `routes.rb` file
- A controller for random messages and one action created
- Database migration for messages table exists

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**